plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerFactory.groovy

```diff
@@ -39,13 +39,25 @@ class TowerFactory implements TraceObserverFactory {
     Collection<TraceObserver> create(Session session) {
         final config = session.config
         Boolean isEnabled = config.navigate('tower.enabled') as Boolean || env.get('TOWER_WORKFLOW_ID')
-        String endpoint = config.navigate('tower.endpoint') as String
-        Duration requestInterval = config.navigate('tower.requestInterval') as Duration
-        Duration aliveInterval = config.navigate('tower.aliveInterval') as Duration
 
         if( !isEnabled )
             return Collections.emptyList()
 
+        final result = new ArrayList(1)
+        // create the tower client
+        final tower = createTowerClient(session, config)
+        result.add(tower)
+        // create the logs checkpoint
+        if( env.containsKey('NXF_CLOUDCACHE_PATH') )
+            result.add( new LogsCheckpoint() )
```

Comment:
Why not enable the logs checkpointing all the time?

Reply:
This is mostly needed when logs are created in the container and need to be copied in the remote work dir (to be accessed by Tower). it's not required in a normal usage scenario